### DISCUSSION TITLE
Increase Nginx proxy-timeouts

### DISCUSSION THREAD

**Discussion by niklasfrick** [guerzon/vaultwarden#128] [Ideas]
I propose increasing the proxy-send-timeout and proxy-read-timeout values in our Nginx configuration to 24 hours (86400). This change would align with the recommendations in the Ingress Nginx documentation for WebSocket support: https://kubernetes.github.io/ingress-nginx/user-guide/miscellaneous/#websockets

Currently, the documentation suggests setting these timeouts to more than one hour (3600 seconds). However, extending this to 24 hours would provide a benefit.

Proposed configuration (ingress.yaml):

```
nginx.ingress.kubernetes.io/proxy-send-timeout: "86400"
nginx.ingress.kubernetes.io/proxy-read-timeout: "86400"
```

This change would enable our application to handle scenarios like revoking access permissions for users, ensuring the changes take effect even after extended periods of inactivity.
Thoughts on this approach? Are there any concerns about potential drawbacks or resource implications we should consider?

**Comment by ismaildasci** [comment]
Hey! 👋

Increasing the `proxy-send-timeout` and `proxy-read-timeout` values to 24 hours (86400 seconds) could indeed benefit long-lived WebSocket connections, especially for scenarios requiring extended inactivity handling, like revoking access permissions. Here are some things to consider about this approach:

### Potential Benefits
- **Improved User Experience**: For applications that require long-running connections, this change would help maintain session stability without frequent reconnections.
- **Alignment with Documentation**: This aligns with the **Ingress Nginx** recommendations for WebSocket support, which suggests setting timeouts higher than one hour.

### Potential Drawbacks
- **Resource Consumption**: Keeping connections open for extended periods can consume server resources (memory and CPU), especially if there are numerous inactive or idle connections. It’s worth considering if your infrastructure can handle this, particularly under high loads.
- **Unresponsive Clients**: A long timeout could lead to situations where stale or unresponsive client connections occupy resources unnecessarily. To mitigate this, you might want to implement connection health checks if possible.

### Suggested Approach
If your infrastructure can support it, extending to 24 hours could be reasonable for your use case. To balance resource usage, you might consider monitoring connection health and resource impact initially to ensure performance remains optimal.

Let us know if you’d like further advice on specific use cases or monitoring tools for tracking the impact of this change! 😊
